pkglistgen/tool.py

```diff
@@ -431,15 +431,17 @@ def create_weakremovers(self, target, target_config, directory, output):
 
                 pool.createwhatprovides()
 
+                accepted_archs = set(self.all_architectures)
```

Comment:
Can this possibly have issues with `i686` vs `i586` (and similar flavors)?

Reply:
we can swap the arch replacement with the check